src/main/java/com/pusher/client/Pusher.java

```diff
@@ -284,6 +287,32 @@ public PrivateChannel subscribePrivate(final String channelName, final PrivateCh
         return channel;
     }
 
+
+    /**
+     * Subscribes to a {@link com.pusher.client.channel.PrivateEncryptedChannel} which
+     * requires authentication.
+     *
+     * @param channelName The name of the channel to subscribe to.
+     * @param listener A listener to be informed of both Pusher channel protocol events and subscription data events.
+     * @param eventNames An optional list of names of events to be bound to on the channel. The equivalent of calling {@link com.pusher.client.channel.Channel#bind(String, SubscriptionEventListener)} one or more times.
+     * @return A new {@link com.pusher.client.channel.PrivateEncryptedChannel} representing the subscription.
+     * @throws IllegalStateException if a {@link com.pusher.client.Authorizer} has not been set for the {@link Pusher} instance via {@link #Pusher(String, PusherOptions)}.
+     */
+    public PrivateEncryptedChannel subscribePrivateEncrypted(
+            final String channelName,
+            final PrivateEncryptedChannelEventListener listener,
+            final String... eventNames) {
+
+        throwExceptionIfNoAuthorizerHasBeenSet();
+
+        final PrivateEncryptedChannelImpl channel = factory.newPrivateEncryptedChannel(
```

Comment:
I'm aware it seems an existing pattern in this file but still it may be worth sharing my thoughts on that as potential improvement to consider going forward. I think putting `final` for variables in such simple scope adds more noise and confusion than useful information or actually needed protection. The feedback applies to the params on this method as well.

Reply:
As a long time Java developer, I had my IDE set up to automatically add `final` to any variable which wasn't mutated on saving a file. Java is verbose, but we should still prefer to use what it has.

As you get used to reading it, it does make mutable variables stick out nicely.

Reply:
I didn't refer to the cost of writing ;) I meant the cost of decreased SNR when reading.

Reply:
I think the pertinent sentence there is getting used to reading Java again 🙃